lti_consumer/tests/unit/plugin/test_views_lti_ags.py

```diff
@@ -0,0 +1,270 @@
+                    'resourceId': 'test',
+                    'scoreMaximum': 100,
+                    'label': 'test label',
+                    'tag': '',
+                    'resourceLinkId': self.xblock.location,
+                    'startDateTime': None,
+                    'endDateTime': None,
+                }
+            ]
+        )
+
+    def test_lti_ags_retrieve(self):
+        """
+        Test the LTI AGS retrieve endpoint.
+        """
+        self._set_lti_token('https://purl.imsglobal.org/spec/lti-ags/scope/lineitem.readonly')
+
+        # Create LineItem
+        line_item = LtiAgsLineItem.objects.create(
+            lti_configuration=self.lti_config,
+            resource_id="test",
+            resource_link_id=self.xblock.location,
+            label="test label",
+            score_maximum=100
+        )
+
+        # Retrieve & check
+        lineitem_detail_url = reverse(
+            'lti_consumer:lti-ags-view-detail',
+            kwargs={
+                "lti_config_id": self.lti_config.id,
+                "pk": line_item.id
+            }
+        )
+        response = self.client.get(lineitem_detail_url)
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(
+            response.data,
+            {
+                'id': 'http://testserver/lti_consumer/v1/lti/{}/lti-ags/{}'.format(
+                    self.lti_config.id,
+                    line_item.id
+                ),
+                'resourceId': 'test',
+                'scoreMaximum': 100,
+                'label': 'test label',
+                'tag': '',
+                'resourceLinkId': self.xblock.location,
+                'startDateTime': None,
+                'endDateTime': None,
+            }
+        )
+
+    def test_create_lineitem(self):
+        """
+        Test the LTI AGS LineItem Creation.
+        """
+        self._set_lti_token('https://purl.imsglobal.org/spec/lti-ags/scope/lineitem')
+
+        # Create LineItem
+        response = self.client.post(
+            self.lineitem_endpoint,
+            data=json.dumps({
+                'resourceId': 'test',
+                'scoreMaximum': 100,
+                'label': 'test',
+                'tag': 'score',
+                'resourceLinkId': self.xblock.location,
+            }),
+            content_type="application/vnd.ims.lis.v2.lineitem+json",
+        )
+
+        self.assertEqual(response.status_code, 201)
+        self.assertEqual(
+            response.data,
+            {
+                'id': 'http://testserver/lti_consumer/v1/lti/1/lti-ags/1',
+                'resourceId': 'test',
+                'scoreMaximum': 100,
+                'label': 'test',
+                'tag': 'score',
+                'resourceLinkId': self.xblock.location,
+                'startDateTime': None,
+                'endDateTime': None,
+            }
+        )
+        self.assertEqual(LtiAgsLineItem.objects.all().count(), 1)
```

Comment:
Shouldn't this test also examine the object created?

Reply:
Yes, I've added checks for the created object.